src/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -98,6 +98,25 @@ public void sort() {
         this.header.setSortOrder(SAMFileHeader.SortOrder.coordinate);
     }
 
+    /** Returns a new IntervalList where each interval is padded by the specified amount of bases. */
+    public IntervalList padded(final int before, final int after) {
+        if (before < 0 || after < 0) throw new IllegalArgumentException("Padding values must be >= 0.");
+        final IntervalList padded = new IntervalList(this.getHeader());
+        final SAMSequenceDictionary dict = padded.getHeader().getSequenceDictionary();
+        for (final Interval i : this) {
+            final int start = Math.max(1, i.getStart()-before);
+            final int end   = Math.min(dict.getSequence(i.getContig()).getSequenceLength(), i.getEnd()+after);
```

Comment:
Since we do not check that the contig of an .added() interval is in the dictionary... should you check here to avoid a NPE?